Sources/PrimerSDK/Classes/PCI/Services/RetryHandler.swift

```diff
@@ -0,0 +1,137 @@
+                self.attempt()
+            }
+        } else {
+            self.handleFinalFailure(responseModel: responseModel, error: error)
+        }
+    }
+
+    func handleFinalFailure(responseModel: DispatcherResponseModel, error: Error?) {
+        var errorMessage = "Failed after \(self.retries) retries.\n"
+
+        if self.retries >= self.retryConfig.maxRetries {
+            errorMessage += "Reached maximum retries (\(self.retryConfig.maxRetries)).\nLast error object: \(error?.localizedDescription ?? "nil")"
+        } else if responseModel.metadata.statusCode >= 500 {
+            errorMessage += "Server error: \(responseModel.metadata.statusCode)"
+        } else if let error = error {
+            errorMessage += "Last error object: \(error.localizedDescription)"
+        } else {
+            errorMessage += "Status code: \(responseModel.metadata.statusCode)"
+        }
+        self.logger.debug(message: errorMessage)
+        let retryEvent = Analytics.Event.message(message: errorMessage, messageType: .info, severity: .info)
+        Analytics.Service.record(event: retryEvent)
+
+        self.completion(.failure(PrimerError.missingPrimerConfiguration(userInfo: .errorUserInfoDictionary(),
+                                                                        diagnosticsId: UUID().uuidString)))
+    }
+
+    func attempt() {
+        let startTime = DispatchTime.now()
+        currentTask = urlSession.dataTask(with: request) { data, urlResponse, error in
+
+            let endTime = DispatchTime.now()
+            let requestDuration = Double(endTime.uptimeNanoseconds - startTime.uptimeNanoseconds) / 1_000_000 // Convert to milliseconds
+
+            guard let httpResponse = urlResponse as? HTTPURLResponse else {
+                let error = InternalError.invalidResponse(userInfo: .errorUserInfoDictionary(),
+                                                          diagnosticsId: UUID().uuidString)
+                self.completion(.failure(error))
+                return
+            }
+
+            let metadata = ResponseMetadataModel(responseUrl: httpResponse.url?.absoluteString,
+                                                 statusCode: httpResponse.statusCode,
+                                                 headers: httpResponse.allHeaderFields as? [String: String])
+            let responseModel = DispatcherResponseModel(metadata: metadata, requestDuration: requestDuration, data: data, error: error)
+
+            // Check if the response is successful
+            if (200...299).contains(responseModel.metadata.statusCode) {
+                let successMessage = "Request succeeded after \(self.retries) retries. Status code: \(responseModel.metadata.statusCode)"
+                self.logger.debug(message: successMessage)
+                let retryEvent = Analytics.Event.message(message: successMessage, messageType: .info, severity: .info)
```

Comment:
EventType should be `RETRY_SUCCESS` based on [this RFC](https://www.notion.so/primerio/Analytics-events-Retries-and-caching-f9935591702e4d1ead70c3b7891a43c6?pvs=4#d82d1f64eef44230b0ae06956b318b27).

All logs/events should be checked for alignment

Reply:
Done!